site/src/content/docs/best-practices/templating.mdx

```diff
@@ -0,0 +1,390 @@
+- `toString` - Convert to string
+- `toInt` - Convert to integer
+- `toYaml` - Convert object to YAML
+- `toJson` - Convert object to JSON
+
+## Test Templates Before Deployment
+
+Always preview your templates before deploying:
+
+```bash
+# Preview manifests with default values
+zarf dev inspect manifests --features="values=true"
+
+# Preview with custom values
+zarf dev inspect manifests \
+  --features="values=true" \
+  -f custom-values.yaml
+
+# Preview with inline overrides
+zarf dev inspect manifests \
+  --features="values=true" \
+  --set-values="app.replicas=5,app.environment=staging"
+```
+
+**What to check:**
+- Templates render without errors
+- Output looks as expected
+- No unexpected whitespace
+- Values are correctly interpolated
+- Conditionals evaluate correctly
+
+## Handle Missing Values Gracefully
+
+Protect against missing or nil values:
+
+```yaml
+# ✅ GOOD: Safe against missing values
+{{- if .Values.monitoring }}
+  {{- if .Values.monitoring.enabled }}
+annotations:
+  prometheus.io/scrape: "true"
+  {{- end }}
+{{- end }}
+
+# Alternative: Use default for the entire object
+{{- $monitoring := .Values.monitoring | default dict -}}
+{{- if $monitoring.enabled }}
+annotations:
+  prometheus.io/scrape: "true"
+{{- end }}
+
+# ❌ RISKY: Will fail if monitoring is undefined
+{{- if .Values.monitoring.enabled }}
+annotations:
+  prometheus.io/scrape: "true"
+{{- end }}
+```
+
+## Be Consistent with Indentation
+
+Use the `indent` and `nindent` functions consistently:
+
+```yaml
+# ✅ GOOD: Proper indentation
+apiVersion: v1
+kind: ConfigMap
+data:
+  config.yaml: |
+{{ .Values.app.config | toYaml | indent 4 }}
+
+# ✅ ALSO GOOD: Using nindent (newline + indent)
+apiVersion: v1
+kind: ConfigMap
+data:
+  config.yaml: |-
+{{- .Values.app.config | toYaml | nindent 4 }}
+
+# ❌ AVOID: Inconsistent indentation
+apiVersion: v1
+kind: ConfigMap
+data:
+  config.yaml: |
+    {{ .Values.app.config | toYaml }}
+```
+
+**Choose the right function:**
+- `indent N` - Indent every line by N spaces
+- `nindent N` - Add newline, then indent by N spaces
+
+## Common Errors and Solutions
+
+### Template Parse Errors
+
+**Error:**
+```
+Error: template: :5:14: executing "" at <.Values.app.missing>: map has no entry for key "missing"
+```
+
+**Solution:**
+```yaml
+# Use default for optional values
+value: {{ .Values.app.missing | default "fallback" }}
+
+# Or check if exists first
+{{- if .Values.app.missing }}
+value: {{ .Values.app.missing }}
+{{- end }}
+```
+
+### Type Errors
+
+**Error:**
+```
+Error: wrong type for value; expected string; got int
+```
+
+**Solution:**
+```yaml
+# Convert types explicitly
+value: {{ .Values.app.port | toString | quote }}
+```
+
+### Whitespace Issues
+
+**Problem:** Extra blank lines or spacing in output
+
+**Solution:**
+```yaml
+# Use whitespace control operators
+{{- if .Values.app.debug -}}
+debug: true
+{{- end -}}
+```
+
+### Quoting Problems
+
+**Problem:** YAML parsing fails with boolean-like strings
```

Comment:
I think I understand why we would use the quote function at times, we wouldn't want a type to be interpreted as the incorrect type. Are there situations where YAML parsing would fail?